### PR TITLE
Create payment page

### PR DESCRIPTION
**Pull request by sandrine-ds**

### CODE REVIEW

File: clients/payment/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "dependencies": {
+    "@0no-co/graphql.web": "1.0.4",
+    "@formatjs/intl": "2.9.5",
+    "@juggle/resize-observer": "3.4.0",
+    "@sentry/react": "7.74.1",
+    "@swan-io/boxed": "1.2.0",
+    "@swan-io/chicane": "1.4.1",
+    "@swan-io/lake": "3.3.2",
+    "@swan-io/shared-business": "3.3.2",
+    "@urql/exchange-graphcache": "6.3.3",
```
Comment (zoontek) [inline clients/payment/package.json:29]: A lot of unused dependencies here

File: clients/payment/src/utils/logger.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+const FORCE_DEV_LOGGING = false;
+
+const ENABLED = env.IS_SWAN_MODE && (process.env.NODE_ENV === "production" || FORCE_DEV_LOGGING);
+
+export const initSentry = () => {
+  if (ENABLED) {
+    init({
+      release: env.VERSION,
+      dsn: "https://632023ecffdc437984c7a53bbb3aa7a6@o427297.ingest.sentry.io/5454043",
```
Comment (zoontek) [inline clients/payment/src/utils/logger.ts:16]: Probably not the right value

File: clients/payment/src/utils/projectId.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Option, Result } from "@swan-io/boxed";
```
Comment (zoontek) [inline clients/payment/src/utils/projectId.ts:1]: Unused

File: clients/payment/src/utils/urql.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { GraphCacheConfig } from "../graphql/unauthenticated";
+import { env } from "./env";
+import { requestIdExchange } from "./exchanges/requestIdExchange";
+import { suspenseDedupExchange } from "./exchanges/suspenseDedupExchange";
+import { logBackendError } from "./logger";
+
+export const isCombinedError = (error: unknown): error is CombinedError =>
+  error instanceof CombinedError;
+
+const unauthenticatedCache = cacheExchange<GraphCacheConfig>({
```
Comment (zoontek) [inline clients/payment/src/utils/urql.ts:13]: Probably useless for this project

File: clients/payment/src/utils/urql.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  schema: schema as NonNullable<GraphCacheConfig["schema"]>,
+});
+
+const onError = (error: CombinedError, operation: Operation) => {
+  logBackendError(error, operation);
+};
+
+export const unauthenticatedClient = new Client({
+  url: `${env.BANKING_URL}/api/unauthenticated`,
```
Comment (zoontek) [inline clients/payment/src/utils/urql.ts:22]: Wrong value

File: scripts/graphql/codegen.ts [hunk 6]
```diff
@@ -95,6 +95,23 @@ const config: CodegenConfig = {
...
   generates: {
+    [file("../../clients/payment/src/graphql/unauthenticated.ts")]: {
+      documents: file("../../clients/payment/src/graphql/unauthenticated.gql"),
+      schema: file("./dist/unauthenticated-schema.gql"),
+      plugins: frontendPlugins,
+      config: frontendConfig,
+      documentTransforms: [{ transform: addTypenames }],
+    },
+
+    [file("../../clients/payment/src/graphql/introspection.json")]: {
```
Comment (zoontek) [inline scripts/graphql/codegen.ts:106]: Could ba avoided if not cache

File: clients/payment/src/App.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+  const [selected, setSelected] = useState<ItemId>(items[0].id);
+
+  return (
+    <ErrorBoundary
+      key={route?.name}
+      onError={error => logFrontendError(error)}
+      fallback={({ error }) => <ErrorView error={error} />}
+    >
+      <Suspense fallback={<LoadingView color={colors.gray[50]} />}>
```
Comment (zoontek) [inline clients/payment/src/App.tsx:36]: `colors.gray[50]` is probably too light

File: clients/payment/src/App.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+              <ScrollView>
+                <Box>
+                  <Box>
+                    <LakeLabel
+                      label={t("paymentLink.paymentMethod")}
+                      render={() => (
+                        <SegmentedControl
+                          mode="desktop"
+                          selected={selected}
+                          items={[{ name: "Direct debit", id: "sdd" }]}
```
Comment (zoontek) [inline clients/payment/src/App.tsx:49]: You can reuse your `items` variable

File: clients/banking/src/components/CheckoutPaymentPage.tsx [hunk 9]
```diff
@@ -0,0 +1,3 @@
+export const CheckoutPaymentPage = () => {
+  return <>Test</>;
```
Comment (zoontek) [inline clients/banking/src/components/CheckoutPaymentPage.tsx:2]: Text must be in `<Text>` component or react-native-web will not be happy about it (console.error)

File: clients/payment/src/utils/logger.ts (hunk 2 above)
Comment (sandrine-ds) [reply clients/payment/src/utils/logger.ts:16]: Where can I find the correct value ?

Comment (zoontek) [reply clients/payment/src/utils/logger.ts:16]: You need to create a new app in Sentry. For the moment, comment the `ENABLED` line and hardcode no

```ts
// TODO: Enable Sentry when going in production
// const ENABLED = env.IS_SWAN_MODE && (process.env.NODE_ENV === "production" || FORCE_DEV_LOGGING);
const ENABLED = false;

export const initSentry = () => {
  if (ENABLED) {
    init({
      release: env.VERSION,
      dsn: "TODO",
      environment: env.PAYMENT_URL.includes("preprod")
        ? "preprod"
        : env.PAYMENT_URL.includes("master")
        ? "master"
        : "prod",
      normalizeDepth: 5,
    });
  }
};
```

File: clients/payment/package.json [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  },
+  "browserslist": [
+    ">0.2%",
+    "not op_mini all",
+    "not dead",
+    "not ie <= 11",
+    "safari >= 12"
+  ],
+  "dependencies": {
+    "@0no-co/graphql.web": "1.0.4",
```
Comment (zoontek) [inline clients/payment/package.json:21]: Update all dependencies using their version numbers in `banking` (as I did an update yesterday)

File: clients/payment/src/assets/imgs/logo-swan.svg
Comment (zoontek) [inline clients/payment/src/assets/imgs/logo-swan.svg:1]: Doesn't appear to be used (probably replaced by the `SwanLogo` component), could be deleted.

File: clients/payment/src/utils/projectId.ts (hunk 3 above)
Comment (zoontek) [inline clients/payment/src/utils/projectId.ts:1]: This file is not used, could be deleted (only used on the onboarding project)

File: clients/payment/src/pages/PaymentPage.tsx [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import { PaymentForm } from "../components/PaymentForm";
+
+export const PaymentPage = () => {
+  return <PaymentForm />;
```
Comment (zoontek) [inline clients/payment/src/pages/PaymentPage.tsx:4]: I think `PaymentForm` is not super useful, you can put its content directly here.

File: clients/payment/src/components/PaymentForm.tsx [hunk 12]
```diff
@@ -0,0 +1,369 @@
...
+          />
+        </Box>
+      </Box>
+
+      <Space height={24} />
+
+      {/* Just for tests */}
+      <SwanLogo
+        color={colors.swan[500]}
+        style={{
```
Comment (zoontek) [inline clients/payment/src/components/PaymentForm.tsx:195]: As it's for tests, add an `eslint-disable-next-line the-rule` here

File: clients/payment/src/components/PaymentForm.tsx [hunk 13]
```diff
@@ -0,0 +1,369 @@
...
+      </Field>
+
+      <Space height={24} />
+
+      <Field name="iban">
+        {({ value, valid, error, onChange }) => (
+          <LakeLabel
+            label={t("paymentLink.iban")}
+            render={() => (
+              <LakeTextInput value={value} valid={valid} error={error} onChangeText={onChange} />
```
Comment (zoontek) [inline clients/payment/src/components/PaymentForm.tsx:235]: Don't forget to pass all data to inputs: `onBlur`, `ref` etc, so you have no surprises when starting validating them 🙂

File: clients/payment/src/components/PaymentForm.tsx [hunk 14]
```diff
@@ -0,0 +1,370 @@
...
+                onValueChange={onChange}
+              />
+            )}
+          />
+        )}
+      </Field>
+
+      <Space height={24} />
+
+      <Field name="iban">
```
Comment (ncomont) [inline clients/payment/src/components/PaymentForm.tsx:231]: could you please pass the refs in order to automatically focus the errored fields ?

File: clients/payment/src/components/PaymentForm.tsx [hunk 15]
```diff
@@ -0,0 +1,370 @@
...
+      <LakeText color={colors.gray[700]} style={styles.bottomText}>
+        {t("paymentLink.termsAndConditions", { merchantName })}
+      </LakeText>
+
+      <Space height={32} />
+
+      <Box direction="row" alignItems="baseline">
+        <LakeText>{t("paymentLink.poweredBySwan")}</LakeText>
+        <Space width={4} />
+        <SwanLogo color={colors.swan[500]} style={styles.swanLogo} />
```
Comment (ncomont) [inline clients/payment/src/components/PaymentForm.tsx:366]: shouldn't it be clickable ?

Comment (sandrine-ds) [reply clients/payment/src/components/PaymentForm.tsx:366]: It's not asked to be clickable

**Pull request closed without merging**